src/include/sof/probe/probe.h

```diff
@@ -0,0 +1,80 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+#ifndef __SOF_PROBE_PROBE_H__
+#define __SOF_PROBE_PROBE_H__
+
+#include <ipc/probe.h>
+
+/*
+ * \brief Initialize probes subsystem
+ *
+ * param[in,optional] extraction_probe_dma - DMA associated with extraction
+ *		      In case extraction_probe_dma is NULL, extraction probes
+ *		      are unavailable.
```

Comment:
this is saying that injection probes are always available? Or only available after the init?

Reply:
Extraction is optional to prevent FW from occupying hda capture stream unnecessarily.
As injection dmas are never attached to by default (require explicit probe_dma_set), you are not occupying any streams so these are always "available".

Reply:
@plbossart And initialization of probe subsystem is obligatory for any probe-related actions. Probes have to be explicitly initialized, and deinitialized after use, to prevent unnecessary memory usage.